lib/ecmascript.ts

```diff
@@ -164,6 +153,34 @@ function isZero(value: JSBI): boolean {
   return JSBI.equal(value, ZERO);
 }
 
+function GetMethod<T extends { [s in M]?: (...args: any[]) => unknown }, M extends string & keyof T>(
+  obj: T,
+  methodName: M
+): T[M];
+function GetMethod<T extends { [s in M]?: undefined | ((...args: any[]) => unknown) }, M extends string & keyof T>(
+  obj: T,
+  methodName: M
+): T[M] | undefined {
+  const result = obj[methodName];
+  if (result === undefined) return undefined;
+  if (typeof result !== 'function') throw new TypeError(`'${methodName}' must be a function`);
```

Comment:
Nit: Maybe put this behind DEBUG as this should be type-checked / guaranteed by tsc?